css/src/components/form/form.scss

```
@@ -0,0 +1,238 @@
+/* stylelint-disable max-nesting-depth, selector-max-specificity  */
+
+$form-font-size: $font-size-7;
+$form-font-size-sm: $font-size-8;
+$form-font-size-lg: $font-size-6;
+
+$icon-size: 2.25em;
+
+$icon-color: $secondary-background !default;
+$icon-active-color: $secondary !default;
+
+.control {
+	position: relative;
+	font-size: $form-font-size;
+	text-align: left;
+
+	&.control-icons {
+		.icon {
+			position: absolute;
+			inset-block-start: 0;
+			inset-inline-start: 0;
+			width: $icon-size;
+			height: $icon-size;
+			color: $icon-color;
+			font-size: $form-font-size;
+			pointer-events: none;
+			z-index: $zindex-multi;
+			padding-inline-start: $icon-size;
+		}
+
+		.input,
+		.select select {
+			padding-inline-start: $icon-size;
+		}
+
+		.input,
+		.select {
+			~ .icon {
+				font-size: $form-font-size;
+			}
+
+			&:focus-visible {
+				~ .icon,
+				.icon {
+					color: $icon-active-color;
+				}
+			}
+
+			&.input-sm,
+			&.select-sm {
+				~ .icon,
+				.icon {
+					font-size: $form-font-size-sm;
+				}
+			}
+
+			&.input-lg,
+			&.select-lg {
+				~ .icon,
+				.icon {
+					font-size: $form-font-size-lg;
+				}
+			}
+		}
+
+		&.control-icon-right {
+			.input {
+				padding-inline: $control-padding-horizontal;
+
+				~ .icon {
+					inset-inline-start: unset;
+					inset-inline-end: 0;
+				}
+			}
+
+			.select {
+				~ .icon {
+					inset-inline-start: unset;
+					inset-inline-end: 2.25em;
+				}
+			}
+
+			.input,
+			.select select {
+				padding-inline-end: $icon-size;
+			}
+
+			.input[type='date'] {
+				position: relative;
+
+				&::-webkit-calendar-picker-indicator {
+					position: absolute;
+					width: 1.5em;
+					background: none;
+					color: transparent;
+					inset-inline-end: 0;
+					z-index: $zindex-dropdown;
+				}
+			}
+		}
+	}
+
+	&.is-loading {
+		&::after {
+			@include loader;
+
+			position: absolute !important;
+			inset-inline-end: 0.625em;
+			inset-block-start: 0.625em;
+			z-index: $zindex-multi;
+		}
+
+		&.control-small::after {
+			font-size: $form-font-size-sm;
+		}
+
+		&.control-medium::after {
+			font-size: $form-font-size;
+		}
+
+		&.control-large::after {
+			font-size: $form-font-size-lg;
+		}
+	}
+}
+
+.field {
+	&:not(:last-child) {
+		margin-block-end: 0.75rem;
+	}
+
+	// Modifiers
+
+	&.field-addons {
+		display: flex;
+		justify-content: flex-start;
+
+		.control {
+			&:not(:last-child) {
+				margin-inline-end: -1px;
+			}
+
+			&:first-child {
+				.button,
+				.input,
+				.select select {
+					border-top-right-radius: 0;
+					border-bottom-right-radius: 0;
+				}
+			}
```

Could this be an opportunity to use logical border radius properties?

```suggestion
			&:first-child {
				.button,
				.input,
				.select select {
					border-start-end-radius: 0;
					border-end-end-radius: 0;
				}
			}
```

More info: [`border-start-end-radius`](https://developer.mozilla.org/en-US/docs/Web/CSS/border-start-end-radius) and [`border-end-end-radius`](https://developer.mozilla.org/en-US/docs/Web/CSS/border-end-end-radius)

---

(If this shouldn't change for RTL languages, please feel free to disregard)

---

Updated